common/utils/src/main/scala/org/apache/spark/util/MavenUtils.scala

```diff
@@ -454,9 +465,23 @@ private[spark] object MavenUtils extends Logging {
           md.addExcludeRule(createExclusion(e + ":*", ivySettings, ivyConfName))
         }
         // resolve dependencies
-        val rr: ResolveReport = ivy.resolve(md, resolveOptions)
+        var rr: ResolveReport = ivy.resolve(md, resolveOptions)
         if (rr.hasError) {
-          throw new RuntimeException(rr.getAllProblemMessages.toString)
+          // SPARK-46302: When there are some corrupted jars in the maven repo,
+          // we try to continue without the cache
+          val failedReports = rr.getArtifactsReports(DownloadStatus.FAILED, true)
```

Comment:
My concern is that if the submission machine has a .m2 dir, it will most likely enter the retry process of removing local-m2-cache...